tests/shared.rs

```diff
@@ -38,3 +38,31 @@ fn two_threads() {
 fn many_threads() {
     send_shared_oneshot_and_wait_on_multiple_threads(1000);
 }
+
+#[test]
+fn drop_on_one_task_ok() {
```

Comment:
I think it's somewhat nondeterministic as we can't be sure that the threads are blocking at a particular point. I had to run this a few times to get it to deadlock beforehand.